src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtils.java

```diff
@@ -89,11 +88,11 @@ public static PodTemplate combine(PodTemplate parent, PodTemplate template) {
         String nodeSelector = Strings.isNullOrEmpty(template.getNodeSelector()) ? parent.getNodeSelector() : template.getNodeSelector();
         String serviceAccount = Strings.isNullOrEmpty(template.getServiceAccount()) ? parent.getServiceAccount() : template.getServiceAccount();
         Node.Mode nodeUsageMode = template.getNodeUsageMode() == null ? parent.getNodeUsageMode() : template.getNodeUsageMode();
-        
+
         Set<PodAnnotation> podAnnotations = new LinkedHashSet<>();
-        podAnnotations.addAll(parent.getAnnotations());
         podAnnotations.addAll(template.getAnnotations());
-        
+        podAnnotations.addAll(parent.getAnnotations());
```

Comment:
Maybe I am wrong, but this looks like the precedence favors `parent`, not as you write, `template`:

> changed precedence so that the template will override the parent in inheritance.

Why do you want to change the precedence in the first place? Reading [the documentation](https://github.com/jenkinsci/kubernetes-plugin#pod-template-inheritance), I would expect the `template` to have precedence.

Reply:
I thought the same but actually he is right

https://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html#add(E)

> Adds the specified element to this set if it is not already present.

Reply:
You are right, makes sense when you think about it.